test/controllers/WithdrawProtectionControllerSpec.scala

```diff
@@ -90,57 +104,65 @@ class WithdrawProtectionControllerSpec extends UnitSpec with WithFakeApplication
     psoAdded = false, Seq(), "£1,100,000"
   )
 
+  val fakeRequest = FakeRequest()
+
+  def mockAuthRetrieval[A](retrieval: Retrieval[A], returnValue: A) = {
```

Comment:
I really don't like that we've got this mockAuthRetrieval and/or mockAuthConnector in pretty much every spec. Surely it can be a trait or something simillar?
Have a look at the keystore mock for paye-registration for example: https://github.com/hmrc/paye-registration-frontend/blob/master/test/mocks/KeystoreMock.scala